src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -22,12 +22,32 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// </summary>
     internal sealed partial class NullableWalker : DataFlowPassBase<NullableWalker.LocalState>
     {
+        // PROTOTYPE(NullableReferenceTypes): Reference the collections directly from the NullableWalker
+        // for the containing method rather than copying the collections. (Items are added to the collections
+        // but never replaced so the collections are lazily populated but otherwise immutable.)
+        internal sealed class VariableState
+        {
+            internal readonly ImmutableDictionary<VariableIdentifier, int> VariableSlot;
+            internal readonly ImmutableArray<VariableIdentifier> VariableBySlot;
+            internal readonly ImmutableDictionary<Symbol, TypeSymbolWithAnnotations> VariableTypes;
+
+            internal VariableState(
+                ImmutableDictionary<VariableIdentifier, int> variableSlot,
+                ImmutableArray<VariableIdentifier> variableBySlot,
+                ImmutableDictionary<Symbol, TypeSymbolWithAnnotations> variableTypes)
+            {
+                VariableSlot = variableSlot;
+                VariableBySlot = variableBySlot;
+                VariableTypes = variableTypes;
+            }
+        }
+
         /// <summary>
-        /// The inferred type at the point of declaration of var locals.
+        /// The inferred type at the point of declaration of var locals and parameters.
         /// </summary>
         // PROTOTYPE(NullableReferenceTypes): Does this need to
         // move to LocalState so it participates in merging?
```

Comment:
Comment doesn't make sense. If it is the type at the point of declaration, how could it ever participate in merging? If it needs to participate in merging, it would be the current flow state, not the state at the point of declaration.

Reply:
Removed.

---
In reply to: [193192527](https://github.com/dotnet/roslyn/pull/27283#discussion_r193192527) [](ancestors = 193192527)